### DISCUSSION TITLE
OSD pod Errors: no disk found with OSD ID by osd pod && 1 osds exist in the crush map but not in the osdmap

### DISCUSSION THREAD

**Discussion by purushothamkdr453** [rook/rook#14689] [Q&A]
**ERRORS**

I am running into weird issues with osd pods. osd pods are not starting and failing with crashloopbackoff status. I am getting the following errors when i checked osd pod logs.

```
no disk found with OSD ID by osd pod
````

```
[WRN] OSD_ORPHAN: 1 osds exist in the crush map but not in the osdmap
```

More details below.

**Setup**

`Cluster` - RKE1(Rancher kubernetes engine) K8s cluster 6 nodes(BM Nodes)
`CephCluster` - Created Cephcluster configured to use 5 devices on each node

With the above setup we should ideally have 30 osds. But currently 26 osds are up and 4 osds are failing.

**failed osds**

- OSD-1  
- OSD-27
- OSD-28
- OSD-29

`OSD-28` & `OSD-29` are placed on 5th node of the cluster.
`OSD-1` is placed on 4th node of the cluster.

**osd(1) pod logs error**

```
HEALTH_WARN 1 osds exist in the crush map but not in the osdmap; 14 daemons have recently crashed
[WRN] OSD_ORPHAN: 1 osds exist in the crush map but not in the osdmap
    osd.1 exists in crush map but not in osdmap
[WRN] RECENT_CRASH: 14 daemons have recently crashed
```
**osd(27,28,29) pod logs error(Crashloopbackoff pod)**

```
for _, info in json.load(sys.stdin).items():
        if info['\''osd_id'\''] == 29:
                print(info['\''device'\''], end='\'''\'')
                print('\''found device: '\'' + info['\''device'\''], file=sys.stderr) # log the disk we found to stderr
                sys.exit(0)  # don'\''t keep processing once the disk is found
sys.exit('\''no disk found with OSD ID 29'\'')
'
no disk found with OSD ID 29
+ DEVICE=
```

I started digging more so Iooked at osd prepare logs and figured out some weird things/issues. osd id 27,28 29 are not found in prepare logs and also osd id 1 is found in 2 different prepare logs. More details are below.

**Issue-1 (Same OSD ID is pointing to 2 different devices):**

- OSD id(osd id no: 1)  in prepare log(osd prepare log) is pointing to 2 different devices(/dev/sdb & /dev/sdf). check the image(osd-prepare-node4.png)

**Issue-2 (same osd id are found under 2 different nodes prepare logs):**

- Same OSD ids(osd no 17 & osd no 13) are listed under 2 different osd prepare logs(prepare logs of 2 different nodes). compare the  images(osd-prepare-node4.png & osd-prepare-node5.png).

**osd-prepare-node4.png**

![osd-prepare-node4](https://github.com/user-attachments/assets/e6c60f55-8648-4a23-80f4-508d0a41f8f2)

**osd-prepare-node5.png**

![osd-prepare-node5](https://github.com/user-attachments/assets/e6473883-ba96-4308-895b-ffdcfa86cd0e)

`Note:` I have attached 4th & 5th node prepare logs.
[rook-ceph-osd-prepare-4thnode.txt](https://github.com/user-attachments/files/16885047/rook-ceph-osd-prepare-4thnode.txt)
[rook-ceph-osd-prepare-5thnode.txt](https://github.com/user-attachments/files/16885050/rook-ceph-osd-prepare-5thnode.txt)

Any idea under which circumstance this happen? How to mitigate these issues?

**Comment by travisn** [comment]
Do you perhaps have multipath disks enabled? The same disk seems presented multiple times. Each disk should only be presented once.

**Reply by purushothamkdr453** [reply]
@travisn How or where can i check this i.e multipath disks enabled?

**Reply by travisn** [reply]
`lsblk` on the node should show the type of disk. If mutlipath, it would show `mpath` for the type.

**Comment by sp98** [comment]
Do you know what event led to this situation?  is this a fresh deployment of rook-ceph or a reinstall ? 
```
OSD id(osd id no: 1) in prepare log(osd prepare log) is pointing to 2 different devices(/dev/sdb & /dev/sdf). check the image(osd-prepare-node4.png)
```
Which device was added first?

**Reply by sp98** [reply]
`osd.1 exists in crush map but not in osdmap` would indicate orphaned OSDs. May the OSDs were not cleaned up from last install if its a case of reinstall.

**Reply by sp98** [reply]
Also, what does `ceph osd tree` say ?

**Reply by purushothamkdr453** [reply]
@sp98 ours is fresh deployment. Here is the output of ceph osd tree.

![ceph-osd-tree](https://github.com/user-attachments/assets/eb2b1ff7-b7ab-4cbd-b9de-a08a94b626e9)

**Comment by sp98** [comment]
So looks like there are orphan OSDs but not sure why. Did you remove OSDs or add new one? Or do you know what event led to the current state of the cluster?  
You can remove the orphan OSDs from the crushmap.

**Reply by purushothamkdr453** [reply]
@sp98 I am not sure what exactly has happened before. I just checked with my colleague and he confirmed that someone fired ceph osd rm command on osd1 .Not sure whether this resulted in this state. Meanwhile could you please provide me the command to remove orphan osds from crushmap?

**Reply by microyahoo** [reply]
you can refer to https://docs.ceph.com/en/reef/rados/operations/add-or-rm-osds/#removing-osds-manual

**Reply by purushothamkdr453** [reply]
@microyahoo osd.1 is already up and part of one of the worker node(refere to the previously shared screenshots). Are you suggesting me to take out this osd(i.e osd1 which is already part of node) by following the instructions mentioned in the above link?

If Yes - Let me know what are the next steps after doing this?

**Reply by microyahoo** [reply]
`[WRN] OSD_ORPHAN: 1 osds exist in the crush map but not in the osdmap` means osd.1 was not exists in ceph cluster, maybe you can try the following steps
1. do `ceph osd ls` to make sure osd.1 not exist
2. check the usage of osd.1 via `ceph osd df`
3. remove osd.1 from the crushmap if it has no data
4. recreate it
